generators/server/templates/src/main/java/package/repository/CustomCouchbaseRepository.java.ejs

```diff
@@ -0,0 +1,123 @@
+<%#
+ Copyright 2013-2021 the original author or authors from the JHipster project.
+
+ This file is part of the JHipster project, see https://www.jhipster.tech/
+ for more information.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-%>
+package <%= packageName %>.repository;
+
+<%_ if (searchEngine === 'couchbase') { _%>
+import com.couchbase.client.java.search.SearchOptions;
+import com.couchbase.client.java.search.SearchQuery;
+import com.couchbase.client.java.search.queries.DocIdQuery;
+import com.couchbase.client.java.search.queries.QueryStringQuery;
+import com.couchbase.client.java.search.result.SearchResult;
+import com.couchbase.client.java.search.result.SearchRow;
+import <%= packageName %>.repository.search.SearchCouchbaseRepository;
+<%_ } _%>
+import org.springframework.data.couchbase.core.CouchbaseOperations;
+import org.springframework.data.couchbase.core.mapping.CouchbasePersistentEntity;
+import org.springframework.data.couchbase.repository.query.CouchbaseEntityInformation;
+import org.springframework.data.couchbase.repository.support.SimpleCouchbaseRepository;
+<%_ if (searchEngine === 'couchbase') { _%>
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+<%_ } _%>
+
+import java.io.Serializable;
+<%_ if (searchEngine === 'couchbase') { _%>
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+<%_ } _%>
+
+/**
+ * A custom implementation of {@code CouchbaseRepository}.
+ */
+public class CustomCouchbaseRepository<T, ID extends Serializable> extends SimpleCouchbaseRepository<T, ID><%if (searchEngine === 'couchbase') { %> implements SearchCouchbaseRepository<T, ID><% } %> {
```

Comment:
Do you have any suggestions on how to go about doing this? 
There's also the `queryString` method in the current implementation which parses the request to handle the case of a lookup by id using `DocIdQuery` instead of a `QueryStringQuery` - so I'm not sure what the best way to handle that is if we go with using the `@Query` annotations to use N1QL instead.